### DISCUSSION TITLE
TerminalGuiDesigner renames some controls i changed name on already

### DISCUSSION THREAD

**Discussion by GronbergInfo** [tznind/TerminalGuiDesigner#211] [General]
Hi.

It seems that every time i run TGD and open ./MyView.cs, it loads everything, but renames a few menuItems to its old/default name.
In the TGD, i have trouble/impossible to open the contect menu to access the sub-menuitem to change name and such there.
But i have in VSCode renamed all references to "menuItem" to "bitMenuStopMusic" and the program compiles, next time i edit using TGD and save, this name is reverted (among 2 other items too)

**Comment by tznind** [comment]
Within the designer you can rename menu items by using `Ctrl+R`. 

You can rename in VSCode manually but you will also need to update the `Data` property to the new name.

```csharp
this.miBlahBlahBlah = new Terminal.Gui.MenuItem();
this.miBlahBlahBlah.Data = "miBlahBlahBlah";
```

Let me know if this does not resolve your problem.

The reason it happens is because the designer compiles the .Designer.cs code and then looks at the runtime instances `Data` property to determine the field name.

**Comment by GronbergInfo** [comment]
> Within the designer you can rename menu items by using `Ctrl+R`.
> 
> You can rename in VSCode manually but you will also need to update the `Data` property to the new name.
> 
> ```cs
> this.miBlahBlahBlah = new Terminal.Gui.MenuItem();
> this.miBlahBlahBlah.Data = "miBlahBlahBlah";
> ```
> 
> Let me know if this does not resolve your problem.
> 
> The reason it happens is because the designer compiles the .Designer.cs code and then looks at the runtime instances `Data` property to determine the field name.

Thanks, with the Rename, i could change it in the TGD and it stuck to the name i wanted.
I probably should print out https://raw.githubusercontent.com/tznind/TerminalGuiDesigner/main/src/Keys.yaml on paper.
